static/js/components/forms/ProfileFormFields.js

```diff
@@ -0,0 +1,499 @@
+import React from "react"
+import moment from "moment"
+import { __, find, includes, propEq, range, reverse } from "ramda"
+import { ErrorMessage, Field, FieldArray } from "formik"
+import * as yup from "yup"
+
+import {
+  EMPLOYMENT_EXPERIENCE,
+  EMPLOYMENT_FUNCTION,
+  EMPLOYMENT_INDUSTRY,
+  EMPLOYMENT_LEVEL,
+  EMPLOYMENT_SIZE,
+  HIGHEST_EDUCATION_CHOICES
+} from "../../constants"
+import FormError from "./elements/FormError"
+import { newPasswordFieldValidation } from "../../lib/validation"
+
+const US_ALPHA_2 = "US"
+const CA_ALPHA_2 = "CA"
+
+const US_POSTAL_CODE_REGEX = /[0-9]{5}(-[0-9]{4}){0,1}/
+const CA_POSTAL_CODE_REGEX = /[A-Z][0-9][A-Z] [0-9][A-Z][0-9]/
+const COUNTRIES_REQUIRING_POSTAL_CODE = [US_ALPHA_2, CA_ALPHA_2]
+const COUNTRIES_REQUIRING_STATE = [US_ALPHA_2, CA_ALPHA_2]
+
+const ADDRESS_LINES_MAX = 4
+const seedYear = moment().year()
+
+export const legalAddressValidation = yup.object().shape({
+  profile: yup.object().shape({
```

Comment:
will `yup` let you do something like this?

```js
const requiredString = yup
  .string()
  .trim()
  .required()
```

then you could just supply the `.label` for a lot of these fields, like `name: requiredString.label('Full Name').min(2)`, etc. Could make this code a little more compact and DRY

Reply:
also, should these big validation functions in general live in a separate file? maybe `lib/profile_validation`? not sure, maybe it's more convenient just to have them here to test alongside the component.